astropy/io/votable/tree.py

```diff
@@ -3089,6 +3098,72 @@ def _parse_fits(self, iterator, extnum, config):
 
         return array
 
+    def _parse_parquet(self, iterator, config):
+        """
+        Functionality to parse parquet files that are embedded
+        in VOTables.
+        """
+        # looks like votable already has a "Table" imported.
+        from astropy.table import Table as Table2
+
+        for start, tag, data, pos in iterator:
+            if tag == "STREAM":
+                if start:
+                    warn_unknown_attrs(
+                        "STREAM",
+                        data.keys(),
+                        config,
+                        pos,
+                        ["type", "href", "actuate", "encoding", "expires", "rights"],
+                    )
+                    href = data["href"]
+                    encoding = data.get("encoding", None)
+                else:
+                    break
+
+            else:  # in this case, there is no STREAM, hence no file linked.
+                href = ""
+
+        if not href.startswith(("http", "ftp", "file")):
+            vo_raise(
+                "The vo package only supports remote data through http, ftp or file",
+                self._config,
+                self._pos,
+                NotImplementedError,
+            )
+
+        # Hack to keep windows working
+        try:
+            fd = urllib.request.urlopen(href)
```

Comment:
Should use `urlopen` as context manager so the handler get closed properly upon exit?

https://docs.python.org/3/library/urllib.request.html

Reply:
Maybe? I don't know, it seems to be working as is, so I would not fix what's apparently not broken.